### ISSUE TITLE
Java Haxe Updates for JDK 17

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#1168]
We are seeing errors with Haxe Java projects after the JDK 17 updates (#1124).

**Comment by JoelProminic** [comment]
Checking the test matrix [here](https://github.com/Moonshine-IDE/Moonshine-IDE/issues/1124#issuecomment-1315775679), I see the error was on Windows only.

**Comment by JoelProminic** [comment]
This is probably not directly related to this issue, but @JesseProminic noticed this behavior today for a macOS Java Haxe project, and I could reproduce it on my machine:

    : Error: Unable to access jarfile /Users/-user-/Documents/Flex/temp_workspace/TestHaxeJava_0104/bin/TestHaxeJava_0104-Debug.jar

The actual path is `/Users/-user-/Documents/Flex/temp_workspace/TestHaxeJava_0104/bin/TestHaxeJava_0104.jar`, without the -Debug in the JAR name.  I confirmed that I can run the JAR if I use the correct name:

    TestHaxeJava_0104 % java -jar bin/TestHaxeJava_0104.jar 
    TestHaxeJava_0104.hx:4: Hello, Haxe!

I don't see a debug mode in the project settings, and Build & Debug does not work.   I'd like to update this so that the names match.

**Comment by JoelProminic** [comment]
From @Igazine in https://github.com/Moonshine-IDE/Moonshine-IDE/issues/1124#issuecomment-1371243232 :

>I don't see any errors on Windows with a Haxe project targeting Java.
>
>Moonshine IDE™
>Version 3.3.4, Build 1354, Player: 33.1(Harman)
>Running on: Windows 11
>
>OpenJDK 19.0.1
>
>Moonshine open/build Haxe project (Java output) successful
>
>Before updating OpenJDK to 19 I tested it with 11.0 as well and it was also successful

Note that we are installing JDK 17 for Moonshine SDK Installer, because it looks to be the latest LTS (long term support) release.

**Comment by JoelProminic** [comment]
The Haxe Java project built for me on Windows, but I got the same Build & Run error.   I see @JesseProminic reported that we can work around this by running Project > Build before Project > Build & Run.

**Comment by Igazine** [comment]
This issue is more complex than I originally thought, but the good news is that it's not related to JDK 17, so this specific issue can be closed. I've created a new issue ( #1177 )  for things that need to be fixed/changed.

**Comment by JoelProminic** [comment]
Agreed. Otherwise, the project builds worked fine for me on macOS and Windows.
